src/agent/src/Toolbox/AgentProcessor.php

```diff
@@ -109,6 +126,12 @@ private function handleToolCallsCallback(Output $output): \Closure
                 $result = $event->hasResult() ? $event->getResult() : $this->agent->call($messages, $output->getOptions());
             } while ($result instanceof ToolCallResult);
 
+            --$this->nestingLevel;
+            if ($this->keepToolSources && 0 === $this->nestingLevel) {
+                $result->getMetadata()->add('sources', $this->sources);
+                $this->sources = [];
```

Comment:
shouldn't the resetting be done even when `keepToolSources` is false, to free memory (and avoid leaking them to the next call) ?